src/charm.py

```diff
@@ -91,6 +91,14 @@ class UPFOperatorCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+        if not self.unit.is_leader():
+            # NOTE: In cases where leader status is lost before the charm is
+            # finished processing all teardown events, this prevents teardown
+            # event code from running. Luckily, for this charm, none of the
+            # teardown code is necessary to preform if we're removing the
+            # charm.
+            self.unit.status = BlockedStatus("Scaling is not implemented for this charm")
```

Comment:
I'm not sure about the status, blocked status indicates that an intervention from the user is required.
And let's assume having units 1 and 2, unit 1 is leader and active, unit 2 is non-leader and blocked, if leadership changes, will both of them go to the correct status? 1 being blocked and 2 being active? (I think yes, but asking to confirm :) )
And what if the condition mentioned in the comment changes and we need to run the teardown code when removing the charm?

Reply:
Anyway if this was the conclusion of the ongoing discussion I don't oppose it.

Reply:
I share your concerns. :laughing: I don't like this approach at all, but there doesn't seem to be a great way to handle this.

> if leadership changes, will both of them go to the correct status? 1 being blocked and 2 being active?

Status will change correctly, although admittedly I'm unsure how well this well be handled without any of the set-up code being run. Blocked status is used, I guess, to indicate the the user should just un-scale their application. This was the best pattern @ghislainbourgeois and I could come up with while discussing this issue. Perhaps `Waiting` would be a better status? I'm unsure.

Additionally, we may be able to glean some patterns or a more robust way of handling this from one of these:

- [This discussion in charmhub](https://chat.charmhub.io/charmhub/pl/ze3qfn3yxirq7c4p4dyuo9k4bw)
- This code which looks at a related issue: https://github.com/canonical/mysql-router-k8s-operator/blob/876b2564d7397a1a8573fa989678766220664487/src/lifecycle.py#L128
- Holsim pattern: [Holism: let's make it official?](https://discourse.charmhub.io/t/holism-lets-make-it-official/12059/2) 
- This related issue: [Bug #1979811 “juju emitting `relation_broken` while there are re...” : Bugs : Canonical Juju](https://bugs.launchpad.net/juju/+bug/1979811) 
- The app.planned_units() function
- [This comment from Harry Pidcock](https://chat.charmhub.io/charmhub/pl/9847ybri4pdimqjdozurnk58bw)

I think we could come up with a pretty solid way to handle this situation, but it would be quite a bit of investigation and testing. That being said, I don't know how much time we want to spend digging on this, if we do eventually plan to support scaling.

Reply:
Put like that I see the point of the blocked status. Thanks.